### PR TITLE
fix: return empty objects for user summary API even if no games requested

### PR DESCRIPTION
**Pull request by MSGoodman**
Addresses #2098 by always returning an object(/empty dictionary) rather than an array for the properties `RecentAchievements` and `Awarded`.

This is a super small PR, but people seem to have opinions about doing things this way vs. `json_decode('{}')` or something with `JSON_FORCE_OBJECT`. It's unclear to me what the most idiomatic solution is but I'm open to anything.

### CODE REVIEW

File: public/API/API_GetUserSummary.php [hunk 1]
```diff
@@ -121,9 +121,9 @@
     unset($retVal['LastGame']['RichPresencePatch']);
     unset($retVal['LastGame']['system']);
 } elseif ($recentGamesPlayed === 0) {
-    // if no games were requested, initialize empty arrays for Awarded and RecentAchievements
-    $retVal['Awarded'] = [];
-    $retVal['RecentAchievements'] = [];
+    // if no games were requested, initialize empty objects for Awarded and RecentAchievements
+    $retVal['Awarded'] = json_encode(new stdClass());
+    $retVal['RecentAchievements'] = json_encode(new stdClass());
```
Comment (wescopeland) [inline public/API/API_GetUserSummary.php:126]: This returns strings:

```
GET http://localhost:64000/API/API_GetUserSummary.php?u=WCopeland&g=0&z=WCopeland&y=...
```

```json
	"Awarded": "{}",
	"RecentAchievements": "{}",
```

Comment (MSGoodman) [reply public/API/API_GetUserSummary.php:126]: This is hilarious and dumb, I didn't realize, thank you for catching this. Will fix.

Comment (Jamiras) [reply public/API/API_GetUserSummary.php:126]: Consider using `ArrayObject` as was done for #484: https://github.com/Jamiras/RAWeb/blob/fc2e030c13e460930d47769d9807e125dbb3b123/public/API/API_GetGameInfoAndUserProgress.php#L59-L61

Comment (MSGoodman) [reply public/API/API_GetUserSummary.php]: Thank you very much, that seems to have done it! Still picking up some of the quirks of PHP!

**Review by Jamiras — APPROVED** [review]

**Review by wescopeland — APPROVED** [review]

**Pull request merged**